File: 1sttest.c
//1sttest.c msp 210104
//Simple c-code for printing command-line parameters
#include <stdio.h>

int main(int argc, char *argv[]) {
  int i;
  char **sp = argv;
  
  for (i = 0; i < argc; i++)
    printf("argv[%d] = %s\n", i, *sp++);
  
  puts("Ready!");
  
  return 0;
}

File: 1sttest.pl
# 1sttest.pl msp 210104
# Simple perl-code for printing command line arguments
use warnings;
use strict;

my $i;

# program name in $0 not in $ARGV[0]
print "Program name: $0\n";

# c-style code
for ($i = 0; $i <= $#ARGV; $i++) {
  printf("argv[%d] = %s\n", $i, $ARGV[$i]);
}

print "\nAnother approach:\n";

# more perl-style
$i = 0;
foreach my $argv (@ARGV) {
  printf("argv[%d] = %s\n", $i++, $argv);
}

print "Ready!\n";
